File: applications/bank-account-sink/src/main/java/com/vmware/bank/account/sink/consumer/BankAccountConsumer.java
package com.vmware.bank.account.sink.consumer;

import com.vmware.financial.open.banking.account.service.AccountService;
import com.vmware.financial.open.banking.domain.account.BankAccount;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

import java.util.function.Consumer;

@Component
@RequiredArgsConstructor
@Slf4j
public class BankAccountConsumer implements Consumer<BankAccount> {
    private final AccountService accountService;

    @Override
    public void accept(BankAccount bankAccount) {

        try {
            log.info("save {} ", bankAccount);
            accountService.saveAccount(bankAccount);
        }
        catch (RuntimeException e)
        {
            log.error("Cannot save account:"+bankAccount,e);
        }
    }
}

File: applications/delete/src/main/kotlin/com/vmware/financial/open/banking/bank/sink/BankAccountConsumer.kt
package com.vmware.financial.open.banking.bank.sink

import com.vmware.financial.open.banking.account.service.AccountService
import com.vmware.financial.open.banking.domain.account.BankAccount
import org.springframework.stereotype.Service
import java.util.function.Consumer

/**
 * @author Gregory Green
 */
@Service
 class BankAccountConsumer(private val service: AccountService) : Consumer<BankAccount> {

    override fun accept(bankAccount: BankAccount) {
        service.createAccount(bankAccount)
    }
}